File: pipeline/source.go
// Copyright 2021 The Tiyo authors
//
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at https://mozilla.org/MPL/2.0/.

package pipeline

// Defines a source data structure type
// At present, this maps to one of:
//   - File
//   - Directory
//   - Stream

// Source : A source data structure
type Source struct {

	// The JointJS ID of this element
	ID string `json:"id"`

	// The name of this element in the pipeline
	Name string `json:"name"`

	// The source type of this element
	Type string `json:"sourcetype"`
}

// NewSource : create a new source object
func NewSource(cell map[string]interface{}) *Source {
	source := Source{}
	if cell["id"] != nil {
		source.ID = cell["id"].(string)
	}

	if cell["name"] != nil {
		source.Name = cell["name"].(string)
	}

	if cell["sourcetype"] != nil {
		source.Type = cell["sourcetype"].(string)
	}

	return &source
}

File: pkg/flow/assets/files/js/collections/source.js
/* Copyright 2021 The Tiyo authors
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

class Source extends Base {
    groupType = false;

    constructor() {
        super('source-element-list', 'sourceProperties', collections.source)
    }

    attributes(view, event, x, y) {
        var element = $('.sourceProperties');
        $('#sourcename').val(view.model.attributes.name);
        element.css({
            "position": "absolute",
            "display": "block",
            "left": x,
            "top": y,
        });

        element.find('.done').click((e) => {
            view.model.attributes.name = $('#sourcename').val();
            view.model.attr()['.label'].text = $('#sourcename').val();

            element.css({
                "display": "none",
            });
            element.find('.done').off('click');
            joint.dia.ElementView.prototype.render.apply(view, arguments);
            pipeline.save();
        });

        element.find('.cancel').click((e) => {
            element.css({
                "display": "none",
            });
            element.find('.cancel').off('click');
        });
    }
}

joint.shapes.container.Source = joint.shapes.devs.Model.extend({
    defaults: joint.util.deepSupplement({
        markup: '<g class="rotatable"><g class="scalable"><image class="body"/></g><text class="label"/><g class="inPorts"/><g class="outPorts"/></g>',
        type: 'container.Source',
        perpendicularLinks: true,

        name: "",
        sourcetype: "",
        position: { x: 50, y: 50 },
        size: { width: 50, height: 50 },
        inPorts: ['a', 'b', 'c'],
        outPorts: ['o'],
        attrs: {
            '.label': {
                text: 'Model',
                'ref-x': 0.5,
                'ref-y': 50,
                'font-size': '10pt',
            },
        },
        ports: {
            groups: {
                'in': {
                    attrs: {
                        '.port-body': {
                            magnet: 'passive',
                            r: 3,
                            stroke: 'green',
                            fill: 'green',
                            'stroke-width': 1,
                            'ref-x': -8,
                        },
                        '.port-label': {
                            'fill-opacity': 0,
                        },
                    }
                },
                'out': {
                    attrs: {
                        '.port-body': {
                            r: 3,
                            stroke: 'red',
                            fill: 'red',
                            'stroke-width': 1,
                            'ref-x': 8,
                        },
                        '.port-label': {
                            'fill-opacity': 0,
                        },
                    }
                }
            }
        },
    }, joint.shapes.devs.Model.prototype.defaults)
});

//# sourceURL=/static/js/collections/source.js
